Rename japanese localization file to ja.json and improve translation

kazubu opened the pull request:
It seems 'jp.json' doesn't work in i18next. I have renamed to 'ja.json'.
Additionally I have some work for improve translation to make it easier to understand.

kazubu commented on locales/ja.json:
```diff
@@ -0,0 +1,79 @@
+  "welcome": {
+    "salutation": "こんにちは！タヌキ開発特製スマホの最新アプリ、<b>Turnip Profit</b>へようこそ。",
+    "description": "このアプリは、無人島のカブ価を毎日予測することができます。そのためには、まず自分でデータを入力することが必要です！",
+    "conclusion": "そうすれば、カブ預言者は<b>魔法</b>のように、あなたのこの一週間のカブ価を予測します。"
```
Ok but how can I translate "the <b>Turnip Prophet</b> app"?
A literal translation is 'カブ予言者'. Is it looks fine?

theRTC204 replied:
Think of it like a brand or proper name. Similar to how you left "Github" untranslated at the bottom of the file. It's fine to translate the words around it "the" and "app" however works best for the language/grammar structure, but just the product name itself should be left untranslated.

Does that make sense?

promarcel replied:
@kazubu We decided today, to remove this translation because we would like have a general uniform name for this app, called the same over all languages belong the domain.

kazubu replied:
Yeah that's why I keep it 'Turnip Profit' here.

kazubu commented on locales/ja.json:
```diff
@@ -0,0 +1,79 @@
+    "fluctuating": "波型",
+    "unknown": "わかりません",
+    "large-spike": "跳ね大型(3期型)",
+    "small-spike": "跳ね小型(4期型)"
```
The prediction has been tried for a long time in Japan and "3期型" and "4期型" are major words to specify large spike and small spike. I think it's better to keep for Japanese players.
e.g. https://twitter.com/tokiruki_akiru/status/289238366602526720

theRTC204 replied:
Ok, that's good context. I agree it would be important to stick to whatever would be the common name for something in the given language.

Are there common names for all four patterns in that case?

kazubu replied:
Thanks. Other name is also translated to major words "波形"(fluctuating) and "ジリ貧型"(decreasing). decreasing pattern has another translate "減少型" in some websites  but most of native speakers could be understand both are same. So I think current translation is fine.

kazubu commented on locales/ja.json:
```diff
@@ -0,0 +1,79 @@
+    },
+    "copy-permalink": "パーマリンクをコピー",
+    "permalink-copied": "パーマリンクがコピーされました！",
+    "reset": "カブ預言者をリセット",
```
Sure

kazubu commented on locales/ja.json:
```diff
@@ -0,0 +1,79 @@
+    }
+  },
+  "textbox": {
+    "description": "複数のカブ価を入力すると、カブ預言者はそれをもとに、可能性のある変動パターンを計算して表示します。",
```
It'll be changed to 'このアプリ' means 'this app'.

kazubu replied:
Sure

theRTC204 commented on locales/ja.json:
```diff
@@ -0,0 +1,79 @@
+    "daisy-mae": "ウリ"
+  },
+  "welcome": {
+    "salutation": "こんにちは！タヌキ開発特製スマホの最新アプリ、<b>Turnip Profit</b>へようこそ。",
```
This should actually be `Turnip Prophet` not `Turnip Profit`.

kazubu replied:
Oh sorry! I'll modify to Turnip Prophet.

kazubu commented:
I have changed as request and rebased to single commit.

```
diff --git a/locales/ja.json b/locales/ja.json
index e497969..c518211 100644
--- a/locales/ja.json
+++ b/locales/ja.json
@@ -3,7 +3,7 @@
     "daisy-mae": "ウリ"
   },
   "welcome": {
-    "salutation": "こんにちは！タヌキ開発特製スマホの最新アプリ、<b>Turnip Profit</b>へようこそ。",
+    "salutation": "こんにちは！タヌキ開発特製スマホの最新アプリ <b>Turnip Prophet</b> へようこそ。",
     "description": "このアプリは、無人島のカブ価を毎日予測することができます。そのためには、まず自分でデータを入力することが必要です！",
     "conclusion": "そうすれば、カブ預言者は<b>魔法</b>のように、あなたのこの一週間のカブ価を予測します。"
   },
@@ -32,7 +32,7 @@
     },
     "copy-permalink": "パーマリンクをコピー",
     "permalink-copied": "パーマリンクがコピーされました！",
-    "reset": "カブ預言者をリセット",
+    "reset": "Turnip Prophetをリセット",
     "reset-warning": "本当に全てをリセットしますか？\n\nリセットしたら後戻りはできませんよ！"
   },
   "weekdays": {
@@ -69,7 +69,7 @@
     }
   },
   "textbox": {
-    "description": "複数のカブ価を入力すると、カブ預言者はそれをもとに、可能性のある変動パターンを計算して表示します。",
+    "description": "複数のカブ価を入力すると、Turnip Prophetはそれをもとに、可能性のある変動パターンを計算して表示します。",
     "development": "このアプリはまだ開発中ですが、より良いデータを提供できるようにがんばっています！",
     "thanks": "このアプリが作成できたのは、タヌキ商店のカブ価パターンを解析した<a href=\"https://twitter.com/_Ninji/status/1244818665851289602?s=20\">Ninji氏の成果</a>のおかげです。ありがとうございます！",
     "support": "疑問·コメント·提案などは、<a href=\"https://github.com/mikebryant/ac-nh-turnip-prices/issues\">Github</a>までお願いします。",
```

kazubu commented:
Sorry, I forgot to rebase master.
